pymeasure/instruments/anritsu/anritsuMS2090A.py

```diff
@@ -0,0 +1,516 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    truncated_range,
+    double_validation_value_and_freq
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class AnritsuMS2090A(Instrument):
+    """Anritsu MS2090A Handheld Spectrum Analyzer."""
+
+    def __init__(self, adapter, **kwargs):
+        """Constructor."""
+        self.analysis_mode = None
+        super().__init__(
+            adapter, name="Anritsu MS2090A Handheld Spectrum Analyzer", **kwargs)
+
+    #############
+    #  Mappings #
+    #############
+
+    ONOFF = ["ON", "OFF"]
+    ONOFF_MAPPING = {True: 'ON', False: 'OFF', 1: 'ON', 0: 'OFF', 'ON': 'ON', 'OFF': 'OFF'}
+    OFFFIRSTREPEAT = ['OFF', 'FIRSt', 'REPeat']
+    SPAMODES = ["SPECtrum", "NRADio", "RTSA", "LTE", "EMFMeter", "PANalyzer"]
+    UNITSFREQ = ['Hz', 'kHz', 'MHz', 'GHz']
+    UNITSFREQ_MAP = []
+
+    ####################################
+    #              GPS                 #
+    ####################################
+
+    gps_full = Instrument.measurement(
+        "FETCh:GPS:FULL?",
+        '''
+        Returns the timestamp, latitude, longitude, altitude, and satellite count of the device.
+        '''
+    )
+
+    gps_all = Instrument.measurement(
+        "FETCh:GPS:ALL?",
+        '''
+        Returns the fix timestamp, latitude, longitude, altitude and information on the sat used.
+        '''
+    )
+
+    gps = Instrument.measurement(
+        ":FETCh:GPS?",
+        '''
+        Returns the timestamp, latitude, and longitude of the device.
+        '''
+    )
+
+    gps_last = Instrument.measurement(
+        ":FETCh:GPS:LAST?",
+        '''
+        Returns the timestamp, latitude, longitude, and altitude of the last fixed GPS result.
+        '''
+    )
+
+    external_current = Instrument.measurement(
+        "BIAS:EXT:CURR?",
+        '''
+        This command queries the actual bias current in A
+        '''
+    )
+
+    ####################################
+    # Spectrum Parameters - Wavelength #
+    ####################################
+
+    frequency_center = Instrument.control(
+        "FREQuency:CENTer?", "FREQuency:CENTer %s",
+        "Sets the center frequency in Hz",
+        validator=double_validation_value_and_freq,

```

Comment:
why do you use your double validator, if you state, that you expect the frequency in Hz?